src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -326,41 +352,36 @@ where
 
         // Install collection state for each bound source.
         for (id, (desc, since)) in bindings {
-            let read_handle = desc.connector.get_read_handle::<T>().await.map_err(|e| {
-                StorageError::ClientError(anyhow!("external persist error: {:?}", e))
-            })?;
-
-            // If we got a persist read handle the since we get from coord is meaningless. It's
-            // most likely `T::minimum()`, which is just the coordinator telling us that it _hopes_
-            // that this is the since. We can go to the source (ha!) of the information.
-            let since = read_handle
-                .as_ref()
-                .map(|read| read.since().clone())
-                .unwrap_or(since);
-
-            let read_handle = read_handle.map(|read| {
-                Box::new(ReadHandleWrapper { read_handle: read })
-                    as Box<dyn CollectionReadHandle<T>>
-            });
+            // TODO(petrosagg): durably record the persist shard we mint here
```

Comment:
Because durably persisting the shard id also implicitly stores the frontiers and data from previous runs and the code is not ready to deal with that. Persisting it durably will have to be done together with the that work